integration-tests/test/conftest.py

```diff
@@ -125,7 +125,7 @@ def testing_context(command_line_options: CommandLineOptions, random_generator:
 testing_context.__test__ = False # type: ignore
 
 
-@pytest.yield_fixture(scope='module')
```

Comment:
I think this change will slow down the execution of tests in the sequential case.

Reply:
You are right! It also would increase the memory which the test would use.

Reply:
This is very hard to accept.  I think majority of the times (esp. on the dev's machines), the tests will be run sequentially.  I was hoping for something that doesn't cause regression in performance in the sequential case.  I'm aware this hunk has been removed but the point still stands for the other similar hunks.

Reply:
I was just doing some tests.
It can be done that way. `xdist` has a param `dist`. If it is set to be `loadscope `, the tests would be groupped by module.